src/repository/note.repository.ts

```diff
@@ -81,4 +81,13 @@ export default class NoteRepository {
   public async getNoteListByUserId(id: number, offset: number, limit: number): Promise<Note[]> {
     return await this.storage.getNoteListByUserId(id, offset, limit);
   }
+
+  /**
+   * Get all notes based on their ids
+   * @param noteIds : list of note ids
+   * @returns an array of notes
+   */
+  public async getNotesByIds(noteIds: NoteInternalId[]): Promise<Note[]> {
```

Comment:
as i remember, note has internal id in it, and internal id of the note should not be exposed, we should return `notePublic[]` i suppose

Reply:
what do you think @neSpecc ? I think that is true, as it should not be exposed in the domain level.

Reply:
all internal layers (store, repo, domain) should operate with internal ids (that is why we have them). 

Conversion to the NotePublic is the task for presentation layer